src/sentry/utils/http.py

```diff
@@ -1,15 +1,24 @@
-from collections import namedtuple
-from typing import Optional
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, NamedTuple, Optional
 from urllib.parse import quote, urljoin, urlparse
 
 from django.conf import settings
+from django.http import HttpRequest
 
 from sentry import options
 
-ParsedUriMatch = namedtuple("ParsedUriMatch", ["scheme", "domain", "path"])
+if TYPE_CHECKING:
```

Comment:
Why is this necessary?

Reply:
there's an import cycle :'( -- the `*` imports in `sentry.models.__init__` suck :(